Option to disable stacking with mouse

szmikler opened the pull request:
I am often stacking windows by mistake when dragging and dropping them using mouse. This PR adds an option to disable Mouse Stacking entirely. However, stacking is still available if activated explicitly with Super+S. One can then move windows between stacks using keyboard shortcuts instead of mouse.

![image](https://user-images.githubusercontent.com/43541355/217563409-55ab1b28-2db3-4e1a-b670-25aa4bdf3169.png)

I believe this might be the answer for the following issues:
* https://github.com/pop-os/shell/issues/965
* https://github.com/pop-os/shell/issues/1491

Commit:
Add toggle to enable/disable stacking with mouse behaviour

Commit:
Merge pull request #1 from gahaalt/toggle-stacking-with-mouse

Add toggle to enable/disable stacking with mouse behaviour

mmstick commented:
You'll have to remove the `.idea` files

Commit:
Rename the option

szmikler commented:
Removed the `.idea` files with a force-push to keep clean history.

Commit:
Remove trailing semicolon

n3m0-22 reviewed (changes requested):
This is causing an issue with tiling. It happens whether the new `Mouse Stacking` is enabled or disabled.

Steps to recreate:
 - enable tiling
 - open two windows (e.g., two terminal windows side by side)
 - create a stack with one
 - add the other to the stack (either my mouse or keyboard)
 - close out both windows in the stack
 - open two windows again
 
 I see a few things happening. Some windows will not tile (e.g., `gnome-terminal`, `calculator`). Some windows maximize (e.g., `nautilus`, `firefox`). For windows that are maximized when un-maximized and closed the active hint still shows. Non-tiled windows will also leave behind an active hint. It tends to be left behind by the last window closed. At this point stacking is also no longer working. Restarting gnome fixes the problem till another stack is created.  

https://user-images.githubusercontent.com/11048214/217618868-c5c9cfe6-d556-4a45-8002-8722b8ffac93.mp4  

https://user-images.githubusercontent.com/11048214/217618882-bd46eba9-8f2d-49b1-a23f-53782642df66.mp4  

In the logs this is new.
```
JS ERROR: TypeError: actor is null
reset_visibility/<@/usr/share/gnome-shell/extensions/pop-shell@system76.com/stack.js:435:21
actor_exec/<@/usr/share/gnome-shell/extensions/pop-shell@system76.com/stack.js:531:17
window_exec@/usr/share/gnome-shell/extensions/pop-shell@system76.com/stack.js:537:17
actor_exec@/usr/share/gnome-shell/extensions/pop-shell@system76.com/stack.js:530:14
reset_visibility@/usr/share/gnome-shell/extensions/pop-shell@system76.com/stack.js:433:18
activate@/usr/share/gnome-shell/extensions/pop-shell@system76.com/stack.js:132:14
on_focused@/usr/share/gnome-shell/extensions/pop-shell@system76.com/extension.js:626:193
signals_attach/</</<@/usr/share/gnome-shell/extensions/pop-shell@system76.com/extension.js:1568:38
run@/usr/share/gnome-shell/extensions/pop-shell@system76.com/extension.js:156:24
wake/<@/usr/share/gnome-shell/extensions/pop-shell@system76.com/executor.js:27:24
```

Output of `journalctl -o cat -n 0 -f "$(which gnome-shell)" | grep -v warning`:  

[shell-log.txt](https://github.com/pop-os/shell/files/10688791/shell-log.txt)

mmstick commented:
The crash is due to #1578 . The fix is in #1580

n3m0-22 commented:
I tested with #1580, and the issues is resolved. I will update when the rest of shell testing is complete.

n3m0-22 reviewed (dismissed):
Looks good. Regression testing passed. Requires #1580.

[shell-testing.txt](https://github.com/pop-os/shell/files/10700915/shell-testing.txt)

jacobgkau commented:
Did we want to check with UX on if `Mouse Stacking` is the best phrasing to use? The other two toggles in this menu start with verbs (`Show Window Titles` and `Show Active Hint`), so `Stack with Mouse` or something like that may be more consistent (and more clear; with no context, `Mouse Stacking` sounds like it's stacking mice.)

jacobgkau commented:
Also, the other toggles in this menu apply whether tiling is enabled or not. This one does nothing when tiling is disabled. Should the toggle be disabled (greyed out) when tiling is disabled to clarify that it doesn't do anything in that state?

Why is this one located in between the two `Show ...` items, separating them?

mmstick commented:
We can place it in the extension preferences

Commit:
Move stacking with mouse setting

Commit:
Undo newline removal

szmikler commented:
I removed this option from panel settings and moved it to extension preferences.

Looks like this:
![image](https://user-images.githubusercontent.com/43541355/218256140-d81bbc96-9a5c-4926-a624-9f3b9da6997e.png)

n3m0-22 reviewed (approved):
Looks good. Testing is still passing. 

Note: The change in #1580 is needed for this to work correctly.

rodhfr commented:
Please make this a little more obvious, i'm new to gnome and the only way I could find this option was reading github issues after searching for hours for this.
